extensions/indexes/range/src/main/java/org/exist/indexing/range/RangeIndexConfigAttributeCondition.java

```diff
@@ -34,293 +34,217 @@
+                RangeIndex.LOG.error("Value conversion error when testing predicate for condition, value: {}", v.toString());
+                RangeIndex.LOG.error(e);
                 return false;
             }
+        };
+    }
 
-            rewrittenOperator = Operator.MATCH;
-            lhe = unwrapSubExpression(func.getArgument(0));
-            rhe = unwrapSubExpression(func.getArgument(1));
-        }
-
-        // find the attribute name and value pair from the predicate to check against
-        // first assume attribute is on the left and value is on the right
-        LocationStep testStep = findLocationStep(lhe);
-        AtomicValue testValue = findAtomicValue(rhe);
-
-        if (testStep == null && testValue == null) {
-            switch (operator) {
-                // the equality operators are commutative so if attribute/value pair has not been found,
-                // check the other way around
-                case EQ, NE -> {
-                    testStep = findLocationStep(rhe);
-                    testValue = findAtomicValue(lhe);
-                }
-                // for ordinal comparisons, attribute and value can also be the other way around in the predicate
-                // but the operator has to be inverted
-                case GT, LT, GE, LE -> {
-                    testStep = findLocationStep(rhe);
-                    testValue = findAtomicValue(lhe);
-                    rewrittenOperator = invertOrdinalOperator(rewrittenOperator);
-                }
-            }
-        }
-
-        if (testStep == null || testValue == null || !rewrittenOperator.equals(operator)) {
-            return false;
-        }
-
-        final QName qname = testStep.getTest().getName();
-        if (qname.getNameType() != ElementValue.ATTRIBUTE || !qname.equals(attribute)) {
-            return false;
-        }
+    private static Predicate<AtomicValue> getNumericTester (final Double value) {
+        return throwingPredicateWrapper(v -> v instanceof NumericValue && v.toJavaObject(Double.class).equals(value));
+    }
 
-        try {
-            if (numericComparison) {
-                return testValue instanceof NumericValue &&
-                        testValue.toJavaObject(Double.class).equals(numericValue);
-            }
-            if (testValue instanceof StringValue) {
-                final String testString = testValue.getStringValue();
-                if (caseSensitive) {
-                    return testString.equals(value);
-                }
-                return testString.equalsIgnoreCase(value);
-            }
-        } catch (XPathException e) {
-            RangeIndex.LOG.error(
-                    "Value conversion error when testing predicate for condition, value: {}", testValue.toString());
-            RangeIndex.LOG.error(e);
+    private static java.util.function.Predicate<AtomicValue> getTester (final String value, final boolean caseSensitive) {
```

Comment:
Here you could remove the fully qualified name `java.util.function.Predicate` as it's already imported.

Reply:
Ah, I missed one :)